ntbs-service/Pages/LabResults/Index.cshtml.cs

```diff
@@ -114,20 +114,33 @@ public async Task<IActionResult> OnGetPotentialMatchDataAsync(string manualNotif
             if (!ValidationService.IsValid(combinedModelPropertyKey))
             {
                 var errorMessage = ModelState[combinedModelPropertyKey].Errors.First().ErrorMessage;
-                return new JsonResult(new { errorMessage });
+                return new JsonResult(new { errorMessage, notificationIdSpecified = true });
+            }
+
+            if (string.IsNullOrWhiteSpace(manualNotificationId))
+            {
+                return new JsonResult(new { notificationIdSpecified = false });
```

Comment:
As mentioned below, I don't think we need this flag - I think it's enough to simply return `new OkResult()` here.